### ISSUE TITLE
Step 17 of FROM_BLINKER_TO_RISCV doesn't work for ULX3S

### ISSUE THREAD

**Issue by samontab** [BrunoLevy/learn-fpga#65]
I was following FROM_BLINKER_TO_RISCV tutorial on a ULX3S, and it all started great but when trying to run step 17 on the board it says:

`BOARDS/run_ulx3s.sh step17.v `

> step17.v:455: ERROR: Unimplemented compiler directive or undefined macro `CPU_FREQ.

I checked and indeed **BOARDS/run_ulx3s.sh** doesn't have CPU_FREQ defined while **BOARDS/run_icestick.sh** has it. 

I tried defining it with some values, but then more issues appeared which I'm not sure how to solve:

> ERROR: Module `\femtoPLL' referenced in module `\Clockworks' in cell `\genblk1.pll' is not part of the design.

Any help would be great.

**Comment by BrunoLevy** [comment]
Thank you for notifying this !
I have pushed a new version (-DULX3S was missing on command line), would you test it ?

**Comment by BrunoLevy** [comment]
(note: it will work until step 22, I still need to do something for SPI flash, will do that shortly...)

**Comment by BrunoLevy** [comment]
Yes, I still need to port different things on the ULX3S. Main thing is the SPI Flash pin multiplexed that needs a special primitive, I will need to insert a layer between the SOC and the ULX3S, now working on doing that without making the whole design too complicated....

**Comment by BrunoLevy** [comment]
- Normally, all the steps up to step 21 (included) work in icarus, verilator and on the ULX3S
   (using run.sh, run_verilator.sh and BOARDS/run_ULX3S.sh), please tell me if it is not the case
- Step 22 and later don't work (because they use SPI Flash for which I need to do something special on the ULX3S and in simulation)
